enterprise/server/invocation_stat_service/invocation_stat_service.go

```diff
@@ -112,7 +104,27 @@ func (i *InvocationStatService) GetTrend(ctx context.Context, req *inpb.GetTrend
 		q.AddWhereClause(fmt.Sprintf("(%s)", roleQuery), roleArgs...)
 	}
 
-	q.AddWhereClause(`updated_at_usec > ?`, timeutil.ToUsec(time.Now().Add(-lookbackWindowDays)))
+	if start := req.GetQuery().GetUpdatedAfter(); start.IsValid() {
+		q.AddWhereClause("updated_at_usec >= ?", timeutil.ToUsec(start.AsTime()))
+	} else {
+		// If no start time specified, respect the lookback window field if set,
+		// or default to 7 days.
+		lookbackWindowDays := 7 * 24 * time.Hour
+		if w := req.GetLookbackWindowDays(); w != 0 {
+			if w < 1 || w > 365 {
+				return nil, status.InvalidArgumentErrorf("lookback_window_days must be between 0 and 366")
+			}
+			lookbackWindowDays = time.Duration(w*24) * time.Hour
+		}
+		q.AddWhereClause("updated_at_usec >= ?", timeutil.ToUsec(time.Now().Add(-lookbackWindowDays)))
+	}
+
+	if end := req.GetQuery().GetUpdatedBefore(); end.IsValid() {
+		q.AddWhereClause("updated_at_usec < ?", timeutil.ToUsec(end.AsTime()))
+	}
+
+	addStatusPredicate(q, req.GetQuery().GetStatus())
```

Comment:
I think this would be clearer if "addStatusPredicate" returned a *query_builder.OrClauses and then you added them to the query here. That way someone reading this code can kind of see the whole query / how it's built in one place.

If we did this, we'd probably also want to call addStatusPredicate buildStatusPredicate instead.

Reply:
sg, went with `toStatusClauses` instead of `buildStatusPredicate` to avoid confusion since we're calling `.Build` on the thing being returned